easybuild/framework/easyblock.py

```diff
@@ -2155,7 +2155,7 @@ def prepare_step(self, start_dir=True, load_tc_deps_modules=True):
         curr_modpaths = curr_module_paths()
         for init_modpath in init_modpaths:
             full_mod_path = os.path.join(self.installdir_mod, init_modpath)
-            if full_mod_path not in curr_modpaths:
+            if os.path.exists(full_mod_path) and full_mod_path not in curr_modpaths:
```

Comment:
Are you sure that this doesn't cause problems when starting from an empty installation.
Consider the situation of a new install, there is just the easybuild-prefix directory, no modules in the HMNS and no module tree, not even $prefix/modules. Bootstrap easybuild into /scratch and use that eb to install foss.
Would this change then cause problems?

Reply:
I'm pretty sure, yes. This is done in the prepare step of each installation. If the path does not (yet) exist, then it is of no use for the current EC. And for the next EC (when the path does exist) this check will return true and it will work as usual.

However I just tested it with no MODULEPATH set at all and successfully installed the full foss/2019b tree from scratch (EB MNS, not HMNS but probably doesn't matter, does it?)

Reply:
It's HMNS I'm worried about, not standard MNS. Since there it matters for modules/all/Core, could you run such a check too?

Reply:
Sure. For reference: Starting with MODULEPATH only containing the EB-devel module in $HOME/..., empty installpath and `eb foss-2019b.eb -r --installpath=/tmp/ebinstall --module-naming-scheme=HierarchicalMNS`

Reply:
ok then, if that works i guess we can move foreward here

Reply:
Yes, currently at "Compiler/GCCcore/8.3.0/Perl/5.30.0", so I'd say it does work

Reply:
Finished successfully